targets/csharp-unityv2/source/Shared/Internal/PlayFabHttp/PlayFabHTTP.cs

```diff
@@ -0,0 +1,337 @@
+            {
+                message = condition + "\n" + stacktrace + UnityEngine.StackTraceUtility.ExtractStackTrace();
+                //logger.Log(message);
+                _logMessageQueue.Enqueue(message);
+            }
+        }
+
+        public static bool IsClientLoggedIn()
+        {
+            return !string.IsNullOrEmpty(_authKey);
+        }
+
+        [Serializable]
+        protected internal class HttpResponseObject
+        {
+            public int code;
+            public string status;
+            public object data;
+        }
+
+        protected internal static PlayFabError GeneratePlayFabErrorGeneric(string message, string stacktrace, object customData = null)
+        {
+            var errorDetails = new Dictionary<string, List<string>>();
+            var sb = new StringBuilder();
+            sb.Append(message);
+            if (!string.IsNullOrEmpty(stacktrace))
+            {
+                sb.Append(" | See stack trace in errorDetails");
+                errorDetails.Add("stacktrace", new List<string>() { stacktrace });
+            }
+            return new PlayFabError()
+            {
+                Error = PlayFabErrorCode.InternalServerError,
+                ErrorMessage = sb.ToString(),
+                ErrorDetails = errorDetails,
+                CustomData = customData ?? new object()
+            };
+        }
+
+        protected internal static PlayFabError GeneratePlayFabError(string json, object customData = null)
+        {
+            //deserialize the error
+            var errorDict = JsonWrapper.DeserializeObject<JsonObject>(json, PlayFabUtil.ApiSerializerStrategy);
+
+            Dictionary<string, List<string>> errorDetails = null;
+            if (errorDict.ContainsKey("errorDetails"))
+            {
+                var ed =
+                    JsonWrapper.DeserializeObject<Dictionary<string, List<string>>>(
+                        errorDict["errorDetails"].ToString());
+                errorDetails = ed;
+            }
+            //create new error object
+            return new PlayFabError
+            {
+                HttpCode = errorDict.ContainsKey("code") ? Convert.ToInt32(errorDict["code"]) : 400,
+                HttpStatus = errorDict.ContainsKey("status")
+                    ? (string)errorDict["status"]
+                    : "BadRequest",
+                Error = errorDict.ContainsKey("errorCode")
+                    ? (PlayFabErrorCode)Convert.ToInt32(errorDict["errorCode"])
+                    : PlayFabErrorCode.ServiceUnavailable,
+                ErrorMessage = errorDict.ContainsKey("errorMessage")
+                    ? (string)errorDict["errorMessage"]
+                    : string.Empty,
+                ErrorDetails = errorDetails,
+                CustomData = customData ?? new object()
+            };
+        }
+
+        protected internal static void SendErrorEvent(PlayFabError error)
```

Comment:
For the same reason you don't want all this event stuff in settings, I'm not happy with it in PlayFabHttp.
I think it could be broken into its own thing, contained within settings.  (Likewise with logging)